invenio_rdm_migrator/load/postgresql/transactions.py

```diff
@@ -0,0 +1,126 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM-Migrator is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""PostgreSQL COPY load."""
+
+from enum import Enum
+
+from sqlalchemy import create_engine
+from sqlalchemy.orm import Session
+
+from ...logging import Logger
+from ..base import Load
+from .sequences import AlterSequencesMixin
+
+
+class OperationType(str, Enum):
+    """SQL operation type enumeration."""
+
+    INSERT = "C"
+    UPDATE = "U"
+    DELETE = "D"
+
+    def __eq__(self, other):
+        """Equality test."""
+        return self.value == other.upper()
+
+    def __str__(self):
+        """Return its value."""
+        return self.value
+
+
+class Operation:
+    """SQL operation.
+
+    A transaction is composed by one or more operations.
+    """
+
+    def __init__(self, type, obj):
+        """Constructor."""
+        assert isinstance(type, OperationType)
+        self.type = type
+        self.obj = obj
+
+
+class PostgreSQLExecute(Load, AlterSequencesMixin):
+    """PostgreSQL COPY load."""
+
+    def __init__(self, db_uri, table_generator_mapper, **kwargs):
+        """Constructor."""
+        self.db_uri = db_uri
+        self._engine = create_engine(self.db_uri)
+        # an instance of TableGeneratorMapper
+        self.tgm = table_generator_mapper
+
+    def _cleanup(self, db=False):
+        """No cleanup."""
+        logger = Logger.get_logger()
+        logger.debug("PostgreSQLExecute does not implement _cleanup()")
+        pass
+
+    def _prepare(self, entry):
+        """Identify transaction group and generate the corresponding op-data pairs."""
+        _operations = []
+        for op in entry["operations"]:
+            ops = self.tgm.prepare(
+                OperationType(op["op"].upper()), op["table"], op["data"]
+            )
+            if ops:
+                _operations.extend(ops)
+
+        entry["operations"] = _operations
+
+        return entry
+
+    def _update_obj(self, session, obj):
```

Comment:
This is _very_ ugly code. However, it's the way I found to keep consistency on using only ORM and update __all__ attributes of an object.